### DISCUSSION TITLE
Failed to connect to ESP32: Timed out waiting for packet header

### DISCUSSION THREAD

**Discussion by aj5r1gg** [markqvist/Reticulum#391] [Help & Questions]
Trying to flash a new RNode and all appears to be going well and then a fatal error.

The error is as follows:
```
Installer Ready

Ok, that should be all the information we need. Please confirm the following summary before proceeding. In the next step, the device will be flashed and provisioned, so make sure that you are satisfied with your choices.

Serial port     : /dev/ttyS0
Device type     : RNode 820 - 1020 MHz
Platform        : ESP32
Device MCU      : Espressif Systems ESP32
Firmware file   : rnode_firmware_ng21.zip

------------------------------------------------------------------------------

Is the above correct? [y/N] y

[15:04:01] Checking firmware file availability...
[15:04:02] The latest firmware for this board is version 1.67
[15:04:02] Using existing firmware file: rnode_firmware_ng21.zip for version 1.67
[15:04:02] Verifying firmware integrity...
[15:04:02] Decompressing firmware...
[15:04:02] Firmware decompressed
[15:04:02] Flashing RNode firmware to device on /dev/ttyS0
esptool.py v3.1
Serial port /dev/ttyS0
Connecting........_____....._____....._____....._____....._____....._____....._____

A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header
[15:04:26] Error from flasher (2) while writing.
[15:04:26] Some boards have trouble flashing at high speeds, and you can
[15:04:26] try flashing with a lower baud rate, as in this example:
[15:04:26] rnodeconf --autoinstall --baud-flash 115200

``` 
I have tried setting the baud flash at 115200, even at 57600, to no avail. 

How can I make this work?

**Comment by faragher** [comment]
I'd expect the time-out to be an issue of incorrect port, but I'd guess that would have been detected previously. Do you have multiple machines and boards you can test this out with? Is it an issue with all the hardware available?

The error itself seems to be coming from esptool, so it's possible rnodeconf isn't aware of the type of error. 

I'll be honest, between this and the AutoInterface error, I'm quickly out of my depth. I haven't done any real work with the firmware or the interfaces.

**Comment by aj5r1gg** [comment]
Thanks @faragher for always responding. And, my apologies, I should have mentioned that I tried four different USB ports.

**Reply by faragher** [reply]
Did the autoinstall find the device name? Is in the same name when you plug it into each port? I had an issue today where it didn't release the port when the device was removed and reinserted. Otherwise, if the device name is correct, changing the speed doesn't help, and the hardware itself isn't somehow locked, I am very confused.

**Comment by markqvist** [comment]
Are you sure that `/dev/ttyS0` is the right port? That looks like one of the console ports on your machine, not an external port. Depending on what type of board you are trying to flash, you should get ports that looke something like `/dev/ttyUSB0` or '/dev/ttyACM0', if you are on Linux.

You can also look at the output from `sudo dmesg -w` when plugging in the device to the USB port, which will show you hardware events happening in realtime, and will often print the port the new device is attached to.

**Reply by aj5r1gg** [reply]
Hey Mark, thanks for that. When I start the rnodeconf utility, there's only ever one port listed being /dev/ttyS0 and yet I have six USB ports on the machine. I'm confused but this could by my lack of knowledge of linux

**Reply by faragher** [reply]
Could you go to the terminal and type `ls /dev/tty*` ? This should get you a list of devices, similar to this:

```
/dev/tty    /dev/tty14  /dev/tty20  /dev/tty27  /dev/tty33  /dev/tty4   /dev/tty46  /dev/tty52  /dev/tty59  /dev/tty8     /dev/ttyp2  /dev/ttyp9  /dev/ttyS0
/dev/tty0   /dev/tty15  /dev/tty21  /dev/tty28  /dev/tty34  /dev/tty40  /dev/tty47  /dev/tty53  /dev/tty6   /dev/tty9     /dev/ttyp3  /dev/ttypa  /dev/ttyS1
/dev/tty1   /dev/tty16  /dev/tty22  /dev/tty29  /dev/tty35  /dev/tty41  /dev/tty48  /dev/tty54  /dev/tty60  /dev/ttyACM0  /dev/ttyp4  /dev/ttypb  /dev/ttyS2
/dev/tty10  /dev/tty17  /dev/tty23  /dev/tty3   /dev/tty36  /dev/tty42  /dev/tty49  /dev/tty55  /dev/tty61  /dev/ttyAML0  /dev/ttyp5  /dev/ttypc  /dev/ttyS3
/dev/tty11  /dev/tty18  /dev/tty24  /dev/tty30  /dev/tty37  /dev/tty43  /dev/tty5   /dev/tty56  /dev/tty62  /dev/ttyAML2  /dev/ttyp6  /dev/ttypd
/dev/tty12  /dev/tty19  /dev/tty25  /dev/tty31  /dev/tty38  /dev/tty44  /dev/tty50  /dev/tty57  /dev/tty63  /dev/ttyp0    /dev/ttyp7  /dev/ttype
/dev/tty13  /dev/tty2   /dev/tty26  /dev/tty32  /dev/tty39  /dev/tty45  /dev/tty51  /dev/tty58  /dev/tty7   /dev/ttyp1    /dev/ttyp8  /dev/ttypf
```
It's a lot of devices, but what you're looking for are things like the ACM0 line (my RNode). The AMLs are my GPS and . . . you know I don't know what. But usually that, or a ttyUSB are the kind of things you're looking for.

The port itself shouldn't show up this way, it's only once it's connected.

**Reply by aj5r1gg** [reply]
Thank you for that. I tried that and there are no USB devices, ACM0 or anything else listed. Just /tty(0-59, 6-63, 7, 8, 9) and /ttyS(0-31). 

What else can I try?

**Reply by aj5r1gg** [reply]
Going to terminal and typing `lsusb` shows this:
```
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 045e:0745 Microsoft Corp. Nano Transceiver v1.0 for Bluetooth
Bus 001 Device 002: ID 046d:0843 Logitech, Inc. Webcam C930e
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
``` 
So there are clearly USB devices there. 

Still confused. Sorry.

**Reply by faragher** [reply]
I'm confused too, so we're in good company, but I think we're getting somewhere. You have two root hubs, a webcam, and a bluetooth transceiver. No RNode. So either the cable isn't working (I actually built a box once to test if a cable was data or charging. It was that big of a problem) or it's not connecting to the device for some reason. Possibly drivers, but we can work with this.

Fairly certain this is my RNode using lsusb:

`Bus 001 Device 006: ID 1a86:55d4 QinHeng Electronics USB Single Serial`

You can see what's happening when you plug/unplug with the following command:

`tail -f /var/log/dmesg | grep USB`

If you don't see anything, you can run `tail -f /var/log/dmesg` but that might give you too much information.  In either case, that should provide additional error messages if any occur.

**Reply by aj5r1gg** [reply]
Oops, my apologies, some more information... When I connect the RNode to a standard charging battery (one that I used to charge my Pixel phone), the RNode powers up. When I connect the RNode to my computer via USB, the RNode DOES NOT power up.  I have tried all of the USB ports to no avail.

I've looked at my computer and there are six USB 3.0 ports. The RNode is powered via a  Micro USB port or an internal battery (with a JST GH 2pin 1.25mm connector).  I am still waiting on my batteries and JST connectors to arrive.

**Reply by aj5r1gg** [reply]
Thank you @faragher for your response. There were no results when I entered ` tail -f /var/log/dmesg | grep USB`. And when I entered `tail -f /var/log/dmesg` there were a dozen or so results but nothing related to USB.

**Reply by faragher** [reply]
Okay, now I am very confused. If the RNode powers up with a USB power bank, we can assume the cable and connector are functioning. USB ports should provide power without any connectivity (even if at a low current). At this point I'm throwing anything at the wall, but do other devices work in those USB ports?

I am entirely at a loss as to why you can't get power to the RNode.

**Reply by aj5r1gg** [reply]
I can stick any other device, such as webcam or flashdrive, into any USB port and they work fine.

**Reply by faragher** [reply]
I don't even know where to go from here. There's some talk about Ryzen chipsets having USB ports not working, but I'm not even sure how ports that power some USB devices would suddenly not power another. I'm sorry, I have no idea what's going on or where to go from here.

**Reply by aj5r1gg** [reply]
That's cool. Thank you for your assistance. We'll work it out, somehow.
